hudi-flink/src/main/java/org/apache/hudi/sink/compact/CompactionCommitSink.java

```diff
@@ -108,8 +124,15 @@ public void invoke(CompactionCommitEvent event, Context context) throws Exceptio
    * @param events  Commit events ever received for the instant
    */
   private void commitIfNecessary(String instant, Collection<CompactionCommitEvent> events) throws IOException {
-    HoodieCompactionPlan compactionPlan = CompactionUtils.getCompactionPlan(
-        this.writeClient.getHoodieTable().getMetaClient(), instant);
+    HoodieCompactionPlan compactionPlan;
+    if (compactionPlanCache.containsKey(instant)) {
+      compactionPlan = compactionPlanCache.get(instant);
+    } else {
+      compactionPlan = CompactionUtils.getCompactionPlan(
+          this.table.getMetaClient(), instant);
+      compactionPlanCache.put(instant, compactionPlan);
```

Comment:
What is the benefit to cache the compaction plan ? Personally i prefer less cached items, the less states the better.

Reply:
When the number of filegroups contained in the compression plan is relatively large, it will be very expensive to createTable and read the compression plan for each event received.

Reply:
Got your idea, can we use `computeIfAbsent` instead ?

Reply:
Ok, it's is a good idea.